dev/com.ibm.ws.jca_fat_workcontext/test-bundles/threadingtestbundle/src/test/bundle/threading/ThreadFactoryImpl.java

```diff
@@ -0,0 +1,50 @@
+package test.bundle.threading;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ThreadFactoryImpl implements ThreadFactory {
+    private final AtomicInteger createdThreadCount = new AtomicInteger();
+    private final ThreadGroup threadGroup;
+    private final ClassLoader contextClassLoader;
+
+    public ThreadFactoryImpl() {
+        this.threadGroup = AccessController.doPrivileged(new PrivilegedAction<ThreadGroup>() {
+            @Override
+            public ThreadGroup run() {
+                return new ThreadGroup("testThreadGroup");
+            }
+        });
+        this.contextClassLoader = AccessController.doPrivileged(new PrivilegedAction<ClassLoader>() {
+            @Override
+            public ClassLoader run() {
+                return Thread.currentThread().getContextClassLoader();
+            }
+        });
+    }
+
+    @Override
+    public Thread newThread(final Runnable runnable) {
+        int threadId = createdThreadCount.incrementAndGet();
+        final String name = "com.ibm.ws.threading_fat_ThreadFactoryImpl-thread-" + threadId;
```

Comment:
```suggestion
        final String name = "test.bundle.threading.ThreadFactoryImpl-thread-" + threadId;
```
Or maybe it should have the jca workcontext fat bucket name instead?